apps/hash-frontend/src/blocks/page/mention-view/mention-display.tsx

```diff
@@ -1,156 +1,328 @@
-import { systemUserShortname } from "@local/hash-isomorphic-utils/environment";
+import { types } from "@local/hash-isomorphic-utils/ontology-types";
+import { extractEntityUuidFromEntityId } from "@local/hash-subgraph";
 import {
-  EntityId,
-  extractEntityUuidFromEntityId,
-  OwnedById,
-} from "@local/hash-subgraph";
-import { FunctionComponent, useMemo } from "react";
+  getEntityRevision,
+  getEntityTypeById,
+  getOutgoingLinkAndTargetEntities,
+  getPropertyTypeById,
+  getRoots,
+} from "@local/hash-subgraph/stdlib";
+import { extractBaseUrl } from "@local/hash-subgraph/type-system-patch";
+import { Box, Popover, styled, Tooltip, Typography } from "@mui/material";
+import { FunctionComponent, useMemo, useRef, useState } from "react";
 
-import { useAccountPages } from "../../../components/hooks/use-account-pages";
 import { useEntityById } from "../../../components/hooks/use-entity-by-id";
-import { useUserOrOrgShortnameByOwnedById } from "../../../components/hooks/use-user-or-org-shortname-by-owned-by-id";
-import { useUsers } from "../../../components/hooks/use-users";
-import { PageIcon } from "../../../components/page-icon";
+import { useGetOwnerForEntity } from "../../../components/hooks/use-get-owner-for-entity";
 import { generateEntityLabel } from "../../../lib/entities";
 import { constructPageRelativeUrl } from "../../../lib/routes";
+import { ArrowUpRightRegularIcon } from "../../../shared/icons/arrow-up-right-regular-icon";
 import { Link } from "../../../shared/ui";
-import { MentionType } from "../create-suggester/mention-suggester";
+import { useEntityIcon } from "../../../shared/use-entity-icon";
+import { Mention } from "../create-suggester/mention-suggester";
```

Comment:
Teechnically this should be in a shared folder (the only folder name that should appear after a `../` is `shared`, although I appreciate we have some legacy buckets like 'components' lying around)